components/calendar/src/islamic.rs

```diff
@@ -1465,4 +1468,42 @@ mod test {
             );
         }
     }
+
+    #[test]
+    fn test_days_in_provided_year_observational() {
+        // -1245 1 1 = -214526 (R.D Date)
+        // 1518 1 1 = 764589 (R.D Date)
+        let x_year = -1245;
+        let y_year = 1518;
+
+        let mut sum_days_in_year = 0;
+        for i in x_year..y_year {
+            let days = IslamicObservational::days_in_provided_year(i);
+            sum_days_in_year += days;
+        }
```

Comment:
optional (also not sure if this works):
```suggestion
        let sum_days_in_year = x_year..y_year
           .map(IslamicObservational::days_in_provided_year)
           .sum();
```

Reply:
Yeah that should work, may need some tweaking to get references to work